packages/services/api/src/modules/organization/providers/organization-member-roles.ts

```diff
@@ -83,19 +88,74 @@ export class OrganizationMemberRoles {
     });
   }
 
-  async getMemberRolesForOrganizationId(organizationId: string) {
+  async getPaginatedMemberRolesForOrganizationId(
+    organizationId: string,
+    args: {
+      first: number | null;
+      after: string | null;
+    },
+  ) {
+    let cursor: { id: string; createdAt: string } | null = null;
+    if (args.after) {
+      cursor = decodeCreatedAtAndUUIDIdBasedCursor(args.after);
+    }
+    const limit = args.first ? (args.first > 0 ? Math.min(args.first, 50) : 50) : 50;
+
     const query = sql`
       SELECT
         ${organizationMemberRoleFields}
       FROM
         "organization_member_roles"
       WHERE
         "organization_id" = ${organizationId}
+        ${
+          cursor
+            ? sql`
+                AND (
+                  (
+                    "created_at" = ${cursor.createdAt}
+                    AND "id" < ${cursor.id}
+                  )
+                  OR "created_at" < ${cursor.createdAt}
+                )
+              `
+            : sql``
+        }
+      ORDER BY
+        "created_at" DESC
+        , "id" DESC
+      LIMIT ${limit + 1}
     `;
 
     const records = await this.pool.any<unknown>(query);
 
-    return records.map(row => MemberRoleModel.parse(row));
+    let edges = records.map(row => {
+      const node = MemberRoleModel.parse(row);
+
+      return {
+        node,
+        get cursor() {
+          return encodeCreatedAtAndUUIDIdBasedCursor(node);
+        },
+      };
+    });
+
+    const hasNextPage = edges.length > limit;
+    edges = edges.slice(0, limit);
+
+    return {
+      edges,
+      pageInfo: {
+        hasNextPage,
+        hasPreviousPage: cursor !== null,
+        get endCursor() {
+          return edges[edges.length - 1]?.cursor ?? '';
+        },
+        get startCursor() {
+          return edges[0]?.cursor ?? '';
```

Comment:
In case you have a preference: Per the spec these cursors can be null. But it seems like they also accept non nulls.

https://relay.dev/graphql/connections.htm#sec-undefined.PageInfo

Reply:
I don't really have a preference here tbh. 🤔